tests/http_status.js

```diff
@@ -146,10 +158,24 @@ function testPage(page, allowedCodes, isFoundLink) {
 
         while (match = linkRegex.exec(rawData)) {
           // found a link
-          if (statusCodes['200'].indexOf(match[1]) == -1 && foundLinks.indexOf(match[1]) == -1) {
-            foundLinkPromises.push(testPage(match[1], ['200', '201', '302'], true));
-            foundLinks.push(match[1]);
+
+          if (match[1].startsWith('#')) {
+            // do not test same-page links
+            continue;
           }
+
+          if (foundLinks.indexOf(match[1]) != -1) {
+            // already found earlier
+            continue;
+          }
+
+          if (Object.keys(statusCodes).some(code => statusCodes[code].indexOf(match[1]) != -1)) {
+            // already defined above
+            continue;
+          }
```

Comment:
What about combining these 3 conditions into a single one by using `||`? Sure, it doesn't make a logical difference, but it is shorter and it seems cleaner to me.